### PR TITLE
Add support for general select expressions.

### PR DESCRIPTION
**Pull request by petermattis**
For example:
  SELECT k,v FROM foo;
  SELECT v+1 FROM foo;
  SELECT 1+1;
  SELECT 1+1 AS two;

### CODE REVIEW

File: sql/driver/driver_test.go [hunk 1]
```diff
@@ -366,6 +367,99 @@ CREATE TABLE t.kv (
...
+
+	testData := []struct {
+		expr     string
+		expected [][]string
+	}{
+		{`1`, [][]string{
+			{"1"},
+			{"1"},
+		}},
+		{`1+1 AS two, 2+2 AS four`, [][]string{
```
Comment (tschottdorf) [inline sql/driver/driver_test.go]: does  `'hello' as str, LENGTH(str) AS len` work?

Comment (petermattis) [inline sql/driver/driver_test.go]: It should. Let me add a test case.

Comment (petermattis) [inline sql/driver/driver_test.go]: Ah, no it doesn't. That doesn't work in postgres either.

Comment (petermattis) [inline sql/driver/driver_test.go]: Both postgres and MySQL report `str` as an unknown column.

Comment (tschottdorf) [inline sql/driver/driver_test.go]: ok, thanks for checking.

**Comment by tschottdorf** [comment]
LGTM

File: sql/select.go [hunk 2]
```diff
@@ -79,7 +120,6 @@ func (p *planner) Select(n *parser.Select) (planNode, error) {
 	// table scan using the primary key.
 
 	// TODO(pmattis): Use a scanNode instead of a valuesNode here.
```
Comment (tamird) [inline sql/select.go]: this comment should have moved with the line below

Comment (petermattis) [inline sql/select.go]: Good point. I'm starting to work on addressing this so I'll fix in that PR.

**Pull request merged**